lua/autorun/ragdollremover.lua

```diff
@@ -11,92 +11,132 @@ local RemoveSpeed = 4000        -- Velocity ragdoll is removed at
 
 local FreezeTime = 3            -- Time body is frozen for
 
-if ( GAMEMODE_NAME == "terrortown" ) then
-        local IsTTT = true
-else
-        local IsTTT = false
-end
+local IsTTT = false
+
+hook.Add("InitPostEntity","CheckForTTT", function() -- This is something I'm not quite sure about, but we may run into an issue that when the file is loaded the TTT gamemode hasn't been loaded. This insures that we check after everything has been loaded.
+	IsTTT = GAMEMODE_NAME == "terrortown" and true or false
+end)
  
-local function SetSubPhysMotionEnabled( ent, enable )
-        if not IsValid( ent ) then return end
-	
-        ent:SetVelocity( vector_origin )
+local entMeta = FindMetaTable("Entity") -- Why? Both ways work but I think the looks cleaner :P
+
+function entMeta:SetSubPhysMotion( enable )
```

Comment:
Like you said it was just my personal preference. I personally think that it just looks cleaner adding it to the Entity MetaTable and calling it that way. I wasn't sure if you would change it or not but I figured I'd throw it in anyway.